src/python/pants/jvm/test/junit.py

```diff
@@ -57,39 +78,58 @@ async def run_junit_test(
     }
 
     reports_dir_prefix = "__reports_dir"
-    reports_dir = f"{reports_dir_prefix}/{field_set.address.path_safe_spec}"
+    reports_dir = f"{reports_dir_prefix}/{request.field_set.address.path_safe_spec}"
 
     # Classfiles produced by the root `junit_test` targets are the only ones which should run.
     user_classpath_arg = ":".join(classpath.root_args())
 
-    process_result = await Get(
-        FallibleProcessResult,
-        Process(
-            argv=[
-                *jdk_setup.args(
-                    bash,
-                    [
-                        *classpath.args(),
-                        *junit_classpath.classpath_entries(toolcp_relpath),
-                    ],
-                ),
-                "org.junit.platform.console.ConsoleLauncher",
-                *(("--classpath", user_classpath_arg) if user_classpath_arg else ()),
-                *(("--scan-class-path", user_classpath_arg) if user_classpath_arg else ()),
-                "--reports-dir",
-                reports_dir,
-                *junit.options.args,
-            ],
-            input_digest=merged_classpath_digest,
-            immutable_input_digests=immutable_input_digests,
-            output_directories=(reports_dir,),
-            append_only_caches=jdk_setup.append_only_caches,
-            env=jdk_setup.env,
-            description=f"Run JUnit 5 ConsoleLauncher against {field_set.address}",
-            level=LogLevel.DEBUG,
-        ),
+    # Cache test runs only if they are successful, or not at all if `--test-force`.
+    cache_scope = (
+        ProcessCacheScope.PER_SESSION if test_subsystem.force else ProcessCacheScope.SUCCESSFUL
     )
 
+    extra_jvm_args: list[str] = []
```

Comment:
Should this default to the value of a `--junit-jvm-args` option? This would allow setting JVM memory options for tests regardless of whether `--test-debug` was in effect.